eventsourcingdjango/recorders.py

```diff
@@ -1,49 +1,83 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+
 from contextlib import contextmanager
 from functools import wraps
 from threading import Lock
-from typing import Any, List, Optional, Type
+from typing import TYPE_CHECKING
 from uuid import UUID
 
 import django.db
 from django.db import models, transaction
+from django.db.backends.signals import connection_created
 from django.db.transaction import get_connection
 from eventsourcing.persistence import (
     AggregateRecorder,
     ApplicationRecorder,
-    DataError,
-    DatabaseError,
-    IntegrityError,
-    InterfaceError,
-    InternalError,
-    NotSupportedError,
     Notification,
     OperationalError,
-    PersistenceError,
     ProcessRecorder,
-    ProgrammingError,
     StoredEvent,
     Tracking,
 )
 
 from eventsourcingdjango.models import NotificationTrackingRecord
 
+if TYPE_CHECKING:
+    from typing import Any, Callable, Dict, Iterator, List, Optional, Type, TypeVar
+
+    from django.utils.connection import ConnectionProxy
+
+    ReturnType = TypeVar("ReturnType")
+    CallableReturnType = Callable[..., ReturnType]
+
+journal_modes: Dict[str, str] = {}
+
+
+class DatabaseError(Exception):
+    pass
+
+
+class DataError(Exception):
+    pass
 
-from django.db.backends.signals import connection_created
 
+class IntegrityError(Exception):
+    pass
 
-journal_modes = {}
 
+class InterfaceError(Exception):
+    pass
 
-def detect_sqlite(connection):
+
+class InternalError(Exception):
+    pass
+
+
+class NotSupportedError(Exception):
+    pass
+
+
+class PersistenceError(Exception):
+    pass
+
+
+class ProgrammingError(Exception):
+    pass
```

Comment:
@johnbywater I've added these exceptions because they weren't in the original `eventsourcing` module. There also some missed imports so there is a question: do you have some unreleased version of `eventsourcing` package while making initial commit in this repo?

Reply:
Yes, I'm currently working from the 'main' branch locally. I needed to adjust the test cases, and didn't want to force myself into releasing a new version of the core library before I was ready, so ended up just developing this again a local version of the core library. Everything is pushed now though. And I will release v9.1 very soon. Just pip install from main for now, if you want to make GH Actions work? Or pip install from a local clone of the core repo, if you have one? Sorry about this.

Reply:
Now, this MR can be reviewed.

Reply:
All looks v great! Thanks so much for doing all of this.